nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EvaluateXPath.java

```diff
@@ -219,6 +239,25 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         }
 
         final ComponentLog logger = getLogger();
+        final DocumentBuilderFactory docBuilderFactory=docBuilderFactoryRef.get();
+        final DocumentBuilder docBuilder;
+        try {
+            docBuilder = docBuilderFactory.newDocumentBuilder();
+
+            if (!context.getProperty(VALIDATE_DTD).asBoolean()) {
+                docBuilder.setEntityResolver(new EntityResolver() {
+                    @Override
+                    public InputSource resolveEntity(String publicId, String systemId) throws SAXException, IOException {
+                        return new InputSource(new StringReader(""));
+                    }
+                });
+            }
+        } catch (ParserConfigurationException e) {
+            logger.error("Configuration error while constructing DocumentBuilder {}", new Object[]{e});
+            session.transfer(flowFiles, REL_FAILURE);
+            return;
```

Comment:
Throwing ProcessException would be more appropriate here, because the ParserConfigurationException is thrown regardless of incoming FlowFiles. By throwing ProcessException, incoming FlowFiles stay in the queue and can be processed after the user have fixed the issue.